pkg/registry/generic/etcd/etcd.go

```diff
@@ -245,13 +248,22 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			}
 			return obj, nil
 		}
+
 		creating = false
+		newVersion, err := e.Helper.ResourceVersioner.ResourceVersion(obj)
```

Comment:
An example is the generic Resize controller.  I'd like to expose Resize in a way that allows it to use the same update logic, but detect whether a size changed in the event of a conflict.  If it didn't, it should allow the update to continue.

----- Original Message -----
> Setting the stage for being able to pass an arbitrary function into Update(),
> to allow Update() to be used from contexts that want to use the AtomicUpdate
> style remerge behavior on specific fields.  The normal Update case is what
> it uses today.
> 
> ----- Original Message -----
> > >  		creating = false
> > > +		newVersion, err := e.Helper.ResourceVersioner.ResourceVersion(obj)
> > 
> > Why is this Update function using the AtomicUpdate pattern? Shouldn't it
> > just
> > use CompareAndSwap?
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/GoogleCloudPlatform/kubernetes/pull/5010/files#r25822475
>